scripts/install_python_prod_dependencies.py

```diff
@@ -175,7 +192,7 @@ def _get_third_party_python_libs_directory_contents():
     """
     direct_url_packages, standard_packages = utils.partition(
         pkg_resources.find_distributions(common.THIRD_PARTY_PYTHON_LIBS_DIR),
-        predicate=lambda dist: dist.has_metadata('direct_url.json'))
+        predicate=_dist_has_meta_data, enumerated=False)
```

Comment:
Why are we adding the `enumerated=False` argument in a typing PR? Ditto below

Reply:
Yeah, handling it with overload

Reply:
done

Reply:
Overload looks great approach to me!